arm/Microsoft.Network/applicationGateways/.bicep/nested_rbac.bicep

```diff
@@ -0,0 +1,12 @@
+param roleAssignment object
```

Comment:
Quoting @MrMCake (who will kindly correct me if I'm wrong): 
_It seems we need to use the resource full scope, as the top-level resource cannot be passed down and used e.g., in a scope: resource fashion to just reference Microsoft.Authorization/roleAssignments_

It looks like we still need something like

```
resource nested_rbac 'Microsoft.Network/applicationGateways/providers/roleAssignments@2020-04-01-preview' = [for principalId in roleAssignment.principalIds: {
  name: '${resourceName}/Microsoft.Authorization/${guid(resourceName, principalId, roleAssignment.roleDefinitionIdOrName)}'
```

Reply:
Correct.

Reply:
changed ;)